src/Model/Profile.php

```diff
@@ -936,4 +937,86 @@ public static function searchProfiles($start = 0, $count = 100, $search = null)
 
 		return ['total' => $total, 'entries' => $profiles];
 	}
+
+	/**
+	 * Migrates a legacy profile to the new slimmer profile with extra custom fields.
+	 * Multi profiles are converted to ACl-protected custom fields and deleted.
+	 *
+	 * @param array $profile One profile array
+	 * @throws \Exception
+	 */
+	public static function migrateFromLegacyProfile(array $profile)
```

Comment:
We're already in the Profile model, so the method name doesn't need to be more verbose than that.
```suggestion
	public static function migrate(array $profile)
```